website/docs/docs/building-a-dbt-project/building-models/python-models.md

```diff
@@ -0,0 +1,560 @@
+
+**Docs:** ["Developer Guide: Snowpark Python"](https://docs.snowflake.com/en/developer-guide/snowpark/python/index.html)
+
+</div>
+
+<div warehouse="Databricks">
+
+**Additional setup:** The `user` field in your [Spark connection profile](spark-profile), usually optional, is required for running Python models.
+
+**Note:** Python models will be created and run as notebooks in your Databricks workspace. The notebook will be created within the personal workspace of the `user` running dbt, and named after the model it is executing. dbt will update the notebooks on subsequent runs of the same model, but it will not delete it—so you can use the notebook for quicker interactive development. Just remember to update the code in your dbt model, based on your in-notebook iteration, before the next `dbt run`.
+
+**Installing packages:** We recommend configuring packages on the interactive cluster which you will be using to run your Python models.
+
+**Docs:**
+- [PySpark DataFrame syntax](https://spark.apache.org/docs/latest/api/python/reference/pyspark.sql/api/pyspark.sql.DataFrame.html)
+- [Databricks: Introduction to DataFrames - Python](https://docs.databricks.com/spark/latest/dataframes-datasets/introduction-to-dataframes-python.html)
+
+</div>
+
+<div warehouse="BigQuery">
+
+The `dbt-bigquery` adapter uses a service called Dataproc to submit your Python models as PySpark jobs. That Python/PySpark code will read from your tables and views in BigQuery, and saves its final result back to BigQuery.
+
+**Additional setup:**
+- Create or use an existing [Cloud Storage bucket](https://cloud.google.com/storage/docs/creating-buckets)
+- Create or use an existing [Dataproc cluster](https://cloud.google.com/dataproc/docs/guides/create-cluster), with the [Spark BigQuery connector initialization action](https://github.com/GoogleCloudDataproc/initialization-actions/tree/master/connectors#bigquery-connectors). (Google recommends copying the action into your own Cloud Storage bucket, rather than using the example version shown in the screenshot below.)
+
+<Lightbox src="/img/docs/building-a-dbt-project/building-models/python-models/dataproc-connector-initialization.png" title="Add the Spark BigQuery connector as an initialization action"/>
+
+Add these attributes to your [BigQuery profile](bigquery-profile):
+- `gcs_bucket`: Storage bucket to which dbt will submit PySpark code, and the Dataproc cluster will stream back execution logs.
+- `dataproc_cluster_name`: Name of Dataproc cluster to use for running Python model (executing PySpark job)
+- `dataproc_region`: GCP region of that Dataproc cluster (e.g. `us-central1`)
+
+The user or service account running dbt needs the following permissions, in addition to those needed for BigQuery ([docs](https://cloud.google.com/dataproc/docs/concepts/iam/iam)):
```

Comment:
saying that all users need to permisssions makes the work clearer than if it's just for a single user...
```suggestion
Any user or service account that will run dbt projects that include `.py` files will need the following permissions in addition to those needed for BigQuery
```